sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/CastSuite.scala

```diff
@@ -819,19 +819,32 @@ class CastSuite extends SparkFunSuite with ExpressionEvalHelper {
   }
 
   test("cast string to boolean") {
-    checkCast("t", true)
+
     checkCast("true", true)
+    checkCast("tru", true)
+    checkCast("tr", true)
+    checkCast("t", true)
     checkCast("tRUe", true)
-    checkCast("y", true)
+    checkCast("    tRue   ", true)
+    checkCast("    tRu   ", true)
     checkCast("yes", true)
+    checkCast("ye", true)
+    checkCast("y", true)
     checkCast("1", true)
+    checkCast("on", true)
 
-    checkCast("f", false)
     checkCast("false", false)
-    checkCast("FAlsE", false)
```

Comment:
Could you add back this since we keep the existing test coverage?

Reply:
done